service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -360,4 +364,32 @@ protected void extractIfMatchVersionParam(final RoutingContext ctx) {
         }
         ctx.next();
     }
+
+    /**
+     * Creates default CORS handler that allows 'POST', 'GET', 'PUT' and 'DELETE' methods for the specified origin.
+     *
+     * @param allowedOrigin The allowed origin pattern.
+     * @return The handler.
+     */
+    protected CorsHandler createDefaultCorsHandler(final String allowedOrigin) {
+        return createCorsHandler(allowedOrigin, EnumSet.of(
+                HttpMethod.POST,
+                HttpMethod.GET,
+                HttpMethod.PUT,
+                HttpMethod.DELETE)
+        );
+    }
+
+    /**
+     * Creates CORS Handler that allows HTTP methods for the specified origin.
+     *
+     * @param allowedOrigin The allowed origin pattern.
+     * @param methods Set of allowed HTTP methods
+     * @return The handler.
+     */
+    protected CorsHandler createCorsHandler(final String allowedOrigin, final Set<HttpMethod> methods) {
+        return CorsHandler.create(allowedOrigin)
+                .allowedMethods(methods)
+                .exposedHeader(HttpHeaders.CONTENT_TYPE.toString());
```

Comment:
what about the *Authorization* header used for authentication of clients?

Reply:
Good catch. Added.